packages/core/src/defaultConfig.ts

```diff
@@ -272,6 +272,14 @@ export const withDefaultConfig = async (
     }
   }
 
+  if (merged.dev?.lazyCompilation === undefined) {
```

Comment:
It would be better if we can update the default config in the `defaultConfig.ts`

https://github.com/web-infra-dev/rsbuild/pull/5582/files#diff-20fef47b581eb119ad418e27d162f0f73b4e85a2977b374b9ee043985d236cf2R52-R54

Reply:
Get👌 Thank you for the detailed explanation!